### PR TITLE
chore: add SDK_SERVER role to api key

### PR DESCRIPTION
**Pull request by cre8ivejp**
Part of #921 

I changed the API Key's role options for easier understanding and added the `SDK_SERVER,` which will be used for server SDKs.
Client and Server SDK will access different data in the server.

All the SDKs and server E2E tests passed using `UNKNOWN` and `SDK_CLIENT` API key roles.
The admin console is also sending the new role value.

```
mysql> select name, role, disabled, created_at, updated_at, environment_namespace from api_key where environment_namespace='yuichi' limit 1;
+--------------+------+----------+------------+------------+-----------------------+
| name         | role | disabled | created_at | updated_at | environment_namespace |
+--------------+------+----------+------------+------------+-----------------------+
| test-api-key |    1 |        0 | 1713328383 | 1713328383 | yuichi                |
+--------------+------+----------+------------+------------+-----------------------+
1 row in set (0.01 sec)
```

### Migration steps

1- Run the following query to change all key roles from `0` to `1`.
**Note**: Currently, we don't have any other role than `SDK` (0).

```
UPDATE api_key SET role = 1
```

2- I'll revert the `FIXME` comments once the table is updated and the E2E tests passed.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -211,14 +211,15 @@ generate-service-token:
...
+# FIXME: Change role to SDK_CLIENT after migration
 .PHONY: create-api-key
 create-api-key:
 	go run ./hack/create-api-key create \
 		--cert=${WEB_GATEWAY_CERT_PATH} \
 		--web-gateway=${WEB_GATEWAY_URL}:443 \
 		--service-token=${SERVICE_TOKEN_PATH} \
 		--name=$$(date +%s) \
-		--role=SDK \
+		--role=UNKNOWN \
```
Comment (cre8ivejp) [inline Makefile:222]: Once the migration is done, I will change the value to `SDK_CLIENT`.

File: pkg/gateway/api/api.go [hunk 2]
```diff
@@ -577,7 +577,9 @@ func (*gatewayService) checkEnvironmentAPIKey(
 	environmentAPIKey *accountproto.EnvironmentAPIKey,
 	role accountproto.APIKey_Role,
 ) error {
-	if environmentAPIKey.ApiKey.Role != role {
+	// TODO: Fix the condition after migration
+	// The role must be UNKNOWN or SDK_CLIENT until the migration is done
+	if environmentAPIKey.ApiKey.Role == accountproto.APIKey_SDK_SERVER {
```
Comment (cre8ivejp) [inline pkg/gateway/api/api.go:582]: As described in the comment, once the migration is done, I will revert the condition.

File: pkg/gateway/api/api_grpc.go [hunk 3]
```diff
@@ -1218,7 +1218,9 @@ func getEnvironmentAPIKeyFromCache(
 }
 
 func checkEnvironmentAPIKey(environmentAPIKey *accountproto.EnvironmentAPIKey, role accountproto.APIKey_Role) error {
-	if environmentAPIKey.ApiKey.Role != role {
+	// TODO: Fix the condition after migration
+	// The role must be UNKNOWN or SDK_CLIENT until the migration is done
+	if environmentAPIKey.ApiKey.Role == accountproto.APIKey_SDK_SERVER {
```
Comment (cre8ivejp) [inline pkg/gateway/api/api_grpc.go:1223]: As described in the comment, once the migration is done, I will revert the condition.

File: ui/web-v2/apps/admin/src/pages/apiKey/index.tsx [hunk 4]
```diff
@@ -271,6 +271,8 @@ export const APIKeyIndexPage: FC = memo(() => {
         createAPIKey({
           environmentNamespace: currentEnvironment.id,
           name: data.name,
+          // TODO: Implement the `role` radio button in the create page
+          role: 1, // SDK_CLIENT
```
Comment (cre8ivejp) [inline ui/web-v2/apps/admin/src/pages/apiKey/index.tsx:275]: This value will be selected in the create page in another PR when I implement the API for the Server SDK.

**Review by Ubisoft-potato — COMMENTED** [review]
LGTM👍!

**Pull request merged**